clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -1513,17 +1513,41 @@ mlir::Value ScalarExprEmitter::emitSub(const BinOpInfo &Ops) {
   if (!mlir::isa<cir::PointerType>(Ops.RHS.getType()))
     return emitPointerArithmetic(CGF, Ops, /*isSubtraction=*/true);
 
-  // Otherwise, this is a pointer subtraction
+  const BinaryOperator *Expr = cast<BinaryOperator>(Ops.E);
+  QualType ElementType = Expr->getLHS()->getType()->getPointeeType();
 
-  // Do the raw subtraction part.
-  //
+  // Check if this is a VLA pointee type.
+  if (const auto *VLA = CGF.getContext().getAsVariableArrayType(ElementType)) {
+    llvm_unreachable("NYI: CIR ptrdiff on VLA pointee");
+  }
   // TODO(cir): note for LLVM lowering out of this; when expanding this into
   // LLVM we shall take VLA's, division by element size, etc.
+  // now we just ensure that all pointers are on the proper address space.
+  mlir::Value LHS = Ops.LHS;
+  mlir::Value RHS = Ops.RHS;
+  mlir::Location loc = CGF.getLoc(Ops.Loc);
+
+  cir::PointerType LHSPtrTy = mlir::dyn_cast<cir::PointerType>(LHS.getType());
```

Comment:
names should be `camelBack`, pre-existing ones you can leave as is, at some point they'd be migrated, but new ones should fit the coding style